app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -688,29 +717,8 @@ export default Component.extend(ComposerUpload, {
     }
 
     if (event.key === "Enter") {
-      const index = parseInt(
-        $(event.target).closest(".button-wrapper").attr("data-image-index"),
-        10
-      );
-      const matchingPlaceholder = this.get("composer.reply").match(
-        IMAGE_MARKDOWN_REGEX
-      );
-      const match = matchingPlaceholder[index];
-      const replacement = match.replace(
-        IMAGE_MARKDOWN_REGEX,
-        `![${$(event.target).val()}|$2$3$4]($5)`
-      );
-
-      this.appEvents.trigger("composer:replace-text", match, replacement);
-
-      const parentContainer = $(event.target).closest(
-        ".alt-text-readonly-container"
-      );
-      const altText = parentContainer.find(".alt-text");
-      const altTextButton = parentContainer.find(".alt-text-edit-btn");
-      altText.show();
-      altTextButton.show();
-      $(event.target).hide();
+      const buttonWrapper = $(event.target).closest(".button-wrapper");
```

Comment:
Can you replace this with vanilla JavaScript? We are trying to use as little jQuery as possible.